### PR TITLE
Feat/56 fede2

### PR DESCRIPTION
**Pull request by fede-rodes**

### CODE REVIEW

File: test/RenPool.test.js [hunk 1]
```diff
@@ -323,11 +332,118 @@ describe('RenPool contract test', function () {
...
+  });
+
+  describe('refundBond', function () {
+
+    it('should refund all stakers', async () => {
+      const registryStoreBalance = await renToken.connect(owner).balanceOf(darknodeRegistryStoreAddr);
+      const aliceBalance = await renToken.connect(alice).balanceOf(alice.address);
+
+      // Lock pool
+      const deposit = POOL_BOND
```
Comment (fede-rodes) [inline test/RenPool.test.js:401]: @acuarica I know you removed all comments, but I think it'll help in the future if someone needs/wants to 'audit' the code. I'd add all comments back into the tests

File: test/RenPool.test.js [hunk 2]
```diff
@@ -26,6 +26,27 @@ const RenToken = require('@renproject/sol/build/testnet/RenToken.json');
 const DarknodeRegistryLogicV1 = require('@renproject/sol/build/testnet/DarknodeRegistryLogicV1.json');
 const DarknodePayment = require('@renproject/sol/build/testnet/DarknodePayment.json');
 
+function base58ToHex(textData) {
```
Comment (acuarica) [inline test/RenPool.test.js:29]: I usually define symbols, _e.g.,_ `functions` and `const` as local as possible. This is to ensure that those symbols are _only_ being used in that scope. Should we place it back to its respective `describe` section?

**Review by acuarica — CHANGES_REQUESTED** [review]

File: contracts/RenPool.sol [hunk 3]
```diff
@@ -94,15 +94,15 @@ contract RenPool {
 	modifier onlyOwnerNodeOperator() {
 		require (
 			msg.sender == owner || msg.sender == nodeOperator,
-			"Caller is not owner/nodeOperator"
+			"RenPool: Caller is not owner nor node operator"
```
Comment (acuarica) [inline contracts/RenPool.sol:97]: Nice! The rationale behind adding the error source, _e.g._, `RenPool`, is to make explicit which contract is throwing the error.

**Review by acuarica — APPROVED** [review]

File: test/RenPool.test.js (hunk 2 above)
Comment (fede-rodes) [reply test/RenPool.test.js:29]: Agree. But in the case of this functions and also increaseMonth and NODE_ID, those are values/functions that are being used in 3 different describe blocks. So in case we scope it to each block we'll triplicate the code

Comment (acuarica) [reply test/RenPool.test.js:29]: Got it. And what about creating a `describe` section to host register/deregister darknode and refund bond? Then we could move these into that `describe` section.

Comment (fede-rodes) [reply test/RenPool.test.js:29]: Ok, let's do that. :+1:

**Review by acuarica — APPROVED** [review]
lgtm

File: test/RenPool.test.js [hunk 4]
```diff
@@ -36,29 +36,30 @@ describe('RenPool contract test', function () {
...
 
-  let owner, nodeOperator, alice, bob;
+  let _, owner, nodeOperator, alice, bob;
   let renPool;
 
   let snapshotID;
 
   before(async () => {
-    [owner, nodeOperator, alice, bob] = await getSigners();
+    [_, owner, nodeOperator, alice, bob] = await getSigners();
```
Comment (acuarica) [inline test/RenPool.test.js:45]: Just for my understanding,
in the commit message https://github.com/Ethernautas/renpool/pull/63/commits/d12edc64f112243dabca9a0ad0006ed008b805c4,
when you mention _lodash signer_ , I'm guessing you're referring to the `_` symbol, but this is not related to _lodash_ the library. Is this correct?

Comment (fede-rodes) [reply test/RenPool.test.js:45]: that's correct, it's not related to the lodash library, just borrowed the name bc it's short but we can change it if you think it's a bit confusing

Comment (acuarica) [reply test/RenPool.test.js:45]: Oh no, I'm fine with the `_`. I was just confused by the `loadash` thing.

**Comment by acuarica** [comment]
How do you usually handle merges? Is the PR author or the reviewer who should merge the PR?

**Comment by acuarica** [comment]
We can ignore the `analysis` check for now.

**Comment by fede-rodes** [comment]
@acuarica ok, let's do the owner to merge the PR
